### DISCUSSION TITLE
redshift driver cannot connect: IncorrectTypeException

### DISCUSSION THREAD

**Discussion by BaronWR** [dbeaver/dbeaver#345] [Q&A] [answered]
Attempting to make a connection to a redshift instance with the standard redshift driver gives the following error:

`SQLException: [Amazon](100021) Error setting default driver property values.
  GeneralException: [Amazon](100021) Error setting default driver property values.
    com.amazon.dsi.exceptions.IncorrectTypeException: IncorrectTypeException`

Other databases seem to work fine.

**Comment by serge-rider** [comment]
Please post the error stacktrace (you can get it in Error Log view)

**Comment by BaronWR** [comment]
I don't see any errors in the error log. This issue occurs when I attempt to connect to redshift.

**Comment by serge-rider** [comment]
What driver (exact version) do you use?

**Comment by WillSidebottom** [comment]
@BaronWr did you ever find a solution to this? I'm having the same problem

**Comment by urbushey** [comment, marked as answer]
@BaronWR @WillSidebottom In case you are still having issues, I think that you may have made the same mistake I initially did when following the Amazon Redshift Getting Started directions [step 5](http://docs.aws.amazon.com/redshift/latest/gsg/rs-gsg-connect-to-cluster.html). 

It says to grab the JDBC URL and copy it into your connection settings. If you just grab the URL, which looks like this:

`jdbc:redshift://xxx.xx.us-east-1.redshift.amazonaws.com:5439/dev`

it includes the `jdbc:redshift://` as well as `:5439/dev` bit. 

You need to take both of those out of your URL because DBeaver connection dialog adds them in. Basically verify that the read-only DBeaver "JDBC URL" matches the Redshift "JDBC URL" at the end of the day because it's easy to mess up.

**Comment by maleckicoa** [comment]
Thanks Urbushey,
 that did the trick!

**Comment by FVDDEV** [comment]
Thanks Urbushey... worked like a wonder !! :)
